components/public-api-server/pkg/idp/idp.go

```diff
@@ -0,0 +1,167 @@
+		keys:                keyCache,
+	}, nil
+}
+
+type Service struct {
+	IssuerBaseURL       string
+	TokenEncryptionCode []byte
+	keys                KeyCache
+}
+
+func (kp *Service) Router() http.Handler {
+	mux := chi.NewRouter()
+	mux.Use(func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			c, _ := httputil.DumpRequest(r, false)
+			log.WithField("req", string(c)).Debug("IDP request")
+			h.ServeHTTP(w, r)
+		})
+	})
+	mux.Handle(oidc.DiscoveryEndpoint, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		keysURL, err := url.JoinPath(kp.IssuerBaseURL, "keys")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		notSupported, err := url.JoinPath(kp.IssuerBaseURL, "not-supported")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		cfg := oidc.DiscoveryConfiguration{
+			Issuer:          kp.IssuerBaseURL,
+			ScopesSupported: op.DefaultSupportedScopes,
+			ResponseTypesSupported: []string{
+				string(oidc.ResponseTypeCode),
+				string(oidc.ResponseTypeIDTokenOnly),
+				string(oidc.ResponseTypeIDToken),
+			},
+			GrantTypesSupported: []oidc.GrantType{
+				oidc.GrantTypeCode,
+				oidc.GrantTypeImplicit,
+			},
+			SubjectTypesSupported: []string{"public"},
+			ClaimsSupported: []string{
+				"sub",
+				"aud",
+				"exp",
+				"iat",
+				"iss",
+				"auth_time",
+				"nonce",
+				"acr",
+				"amr",
+				"c_hash",
+				"at_hash",
+				"act",
+				"scopes",
+				"client_id",
+				"azp",
+				"preferred_username",
+				"name",
+				"family_name",
+				"given_name",
+				"locale",
+				"email",
+			},
+			IDTokenSigningAlgValuesSupported:                   []string{"RS256"},
+			RevocationEndpointAuthMethodsSupported:             []oidc.AuthMethod{oidc.AuthMethodNone},
+			IntrospectionEndpointAuthMethodsSupported:          []oidc.AuthMethod{oidc.AuthMethodNone},
+			IntrospectionEndpointAuthSigningAlgValuesSupported: []string{"RS256"},
+			RequestURIParameterSupported:                       false,
+			AuthorizationEndpoint:                              notSupported,
+			TokenEndpoint:                                      notSupported,
+			IntrospectionEndpoint:                              notSupported,
+			UserinfoEndpoint:                                   notSupported,
+			RevocationEndpoint:                                 notSupported,
+			EndSessionEndpoint:                                 notSupported,
+			JwksURI:                                            keysURL,
+		}
+		err = json.NewEncoder(w).Encode(cfg)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}))
+	mux.Handle("/keys", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		keys, err := kp.keys.PublicKeys(r.Context())
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		_, _ = w.Write(keys)
```

Comment:
Should handle the write error - it returns an error if the underlying request body has been closed by the client in the meantime